core/src/main/java/org/apache/calcite/sql/validate/implicit/TypeCoercionImpl.java

```diff
@@ -382,12 +383,22 @@ protected boolean booleanEquality(SqlCallBinding binding,
     return false;
   }
 
+  @Deprecated @Override public boolean caseWhenCoercion(SqlCallBinding callBinding) {
+    return caseOrEquivalentCoercion(callBinding);
```

Comment:
If you want to keep the original logic, why return the new method?

Reply:
Why is it marked solved without any explanation?

Reply:
I don't think we promise to preserve the original functionality for deprecated methods, just their signature.
But there is an argument to be made for preserving the original functionality too.

Reply:
I understand that if we want to keep the original function, we should keep it intact. Or we can just modify the original method directly.